server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -317,6 +329,39 @@ public String executor() {
         }
     }
 
+    TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) throws Exception {
+        // Read enum
+        switch (extensionRequest.getRequestType()) {
+            case REQUEST_EXTENSION_CLUSTER_STATE:
+                ClusterStateResponse clusterStateResponse = new ClusterStateResponse(
+                    clusterService.getClusterName(),
+                    clusterService.state(),
+                    false
+                );
+                return clusterStateResponse;
+            case REQUEST_EXTENSION_LOCAL_NODE:
+                LocalNodeResponse localNodeResponse = new LocalNodeResponse(clusterService);
+                return localNodeResponse;
+            case REQUEST_EXTENSION_CLUSTER_SETTINGS:
+                ClusterSettingsResponse clusterSettingsResponse = new ClusterSettingsResponse(clusterService);
+                return clusterSettingsResponse;
+            case REQUEST_EXTENSION_ACTION_LISTENER_ON_FAILURE:
+                return handleExtensionActionListenerOnFailureRequest(extensionRequest.getFailureException());
+            default:
+                throw new Exception("Handler not present for the provided request");
```

Comment:
Please use a more narrowly typed exception here.  I'd suggest either `IllegalArgumentException` or `UnsupportedOperationException`.

Other users may want to catch and handle this and this forces them to catch all exceptions.